app/controllers/gene_sequence_assets_controller.rb

```diff
@@ -64,6 +64,34 @@ def destroy
     head :ok
   end
 
+  def rename
```

Comment:
this logic should probably be extracted into a concern, since it's basically the same for all controllers